apps/whale-api/src/module.api/consortium.controller.ts

```diff
@@ -10,6 +10,36 @@ export class ConsortiumController {
     protected readonly cache: SemaphoreCache
   ) {}
 
+  /**
+   *  Gets the transaction history of consortium members.
+   *
+   * @return {Promise<ConsortiumTransactionResponse>}
+    */
+  @Get('/transactions')
+  async getTransactionHistory (
+    @Query() query: { pageIndex?: number, limit?: number, searchTerm?: string }
+  ): Promise<ConsortiumTransactionResponse> {
+    const { pageIndex = 0, limit = 20, searchTerm = undefined } = query
+
+    if (limit > 50 || limit < 1) {
+      throw new ForbiddenException('InvalidLimit')
+    }
+
+    if (searchTerm !== undefined && (searchTerm.length < 3 || searchTerm.length > 64)) {
+      throw new ForbiddenException('InvalidSearchTerm')
+    }
+
+    if (pageIndex < 0) {
+      throw new ForbiddenException('InvalidPageIndex')
+    }
```

Comment:
This bypasses the `PaginationQuery`  validation. Would it be better if you create a new Query class for this purpose of validating in the same file?

Reply:
Added [`9e171c7`](https://github.com/JellyfishSDK/jellyfish/pull/1868/commits/9e171c7dc05223871584a2f372804518aad3dfea)